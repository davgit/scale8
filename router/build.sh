#!/bin/bash

# Clean up...
rm -rf ../ui/build ../ui/out

yarn --cwd ../ui export

rsync -a --delete ../ui/out/ ui-build

docker build -t scale8/router:latest .
