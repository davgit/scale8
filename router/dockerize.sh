#!/bin/bash

docker build -t scale8/router:latest .