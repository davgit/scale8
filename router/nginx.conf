user nginx;

worker_processes    auto;
pcre_jit            on;
error_log           /dev/stdout warn;
include             /etc/nginx/modules/*.conf;

events {
    worker_connections 1024;
}

http {
    include                     /etc/nginx/mime.types;
    default_type                application/octet-stream;
    server_tokens               off;
    client_max_body_size        1m;
    sendfile                    on;

    ssl_protocols               TLSv1.1 TLSv1.2 TLSv1.3;
    ssl_prefer_server_ciphers   on;
    ssl_session_cache           shared:SSL:2m;
    ssl_session_timeout         1h;
    ssl_session_tickets         off;

    gzip_vary                   on;
    keepalive_timeout           65;

    access_log                  /dev/stdout;

    upstream api {
        server      ${API_SERVER:=api:8082};
    }

    upstream edge {
        server      ${EDGE_SERVER:=edge:6080};
    }

    upstream ui {
        server      ${UI_SERVER:=127.0.0.1:3000};
    }

    server {
        listen          80;
        server_name     _;

        root            /opt/www;

        location ~ ^/(graphql|rest|api|ping|preview-revision|auth) {
            proxy_set_header X-Real-IP $$remote_addr;
            proxy_set_header Host $$host;
            proxy_set_header X-Forwarded-For $$proxy_add_x_forwarded_for;
            proxy_pass http://api;
        }

        location ~ ^/(edge) {
            proxy_set_header X-Real-IP $$remote_addr;
            proxy_set_header Host $$host;
            proxy_set_header X-Forwarded-For $$proxy_add_x_forwarded_for;
            proxy_pass http://edge;
        }

        location / {
            ${UI_SERVER+#}${UI_SERVER-try_files $uri $uri.html /index.html;}
            ${UI_SERVER+proxy_set_header X-Real-IP $remote_addr;}
            ${UI_SERVER+proxy_set_header Host $host;}
            ${UI_SERVER+proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;}
            ${UI_SERVER+proxy_pass http://ui;}
        }
    }
}
