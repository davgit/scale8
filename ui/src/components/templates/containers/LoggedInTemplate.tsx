import { FC, MouseEvent, SyntheticEvent } from 'react';
import SideBar from '../../organisms/SideBar';
import { Box, Snackbar } from '@mui/material';
import CancelConfirmDialog from '../../organisms/CancelConfirmDialog';
import { MainDrawer } from '../../organisms/MainDrawer';
import { DialogBaseProps, DialogPageProps } from '../../../types/DialogTypes';
import GQLError from '../../atoms/GqlError';
import InfoDialog from '../../organisms/InfoDialog';
import PageDialog from '../../organisms/PageDialog';
import { useLoggedInState } from '../../../context/AppContext';
import { LoggedInProps } from '../../../containers/global/LoggedInSection';

const LoggedInTemplate: FC<LoggedInProps> = (props: LoggedInProps) => {
    const { templateInteractions } = useLoggedInState();
    const { ask, dispatchDialogAction, dialogState, snackbarError, setSnackbarError } =
        templateInteractions;

    const handleDialogClose = (checkChanges: boolean) => {
        if (checkChanges && dialogState.pageHasChanges) {
            ask('Discard unsaved data?', () => {
                dispatchDialogAction({
                    type: 'close',
                });
            });
        } else {
            dispatchDialogAction({
                type: 'close',
            });
        }
    };

    const drawerProps: DialogBaseProps = {
        open: dialogState.pageComponent !== null && dialogState.type === 'drawer',
        handleDialogClose,
        width: dialogState.pageWidth ?? 432,
        secondaryPageComponent: dialogState.secondaryPageComponent,
    };

    const infoProps: DialogBaseProps = {
        open:
            dialogState.pageComponent !== null &&
            (dialogState.type === 'fullscreen' || dialogState.type === 'info'),
        handleDialogClose,
    };

    const lightboxProps: DialogBaseProps = {
        open: dialogState.pageComponent !== null && dialogState.type === 'lightbox',
        handleDialogClose,
    };

    const dialogPageProps: DialogPageProps = {
        pageRefresh:
            dialogState.refresh ??
            (() => {
                // no refresh
            }),
        handleDialogClose,
        setPageHasChanges: (pageHasChanges: boolean) => {
            dispatchDialogAction({
                type: 'setPageHasChanges',
                payload: { pageHasChanges },
            });
        },
        id: dialogState.contentId ?? '',
        contextId: dialogState.contextId ?? '',
        ids: dialogState.ids ?? [],
        followUp: dialogState.followUp,
        name: dialogState.name,
        readOnly: dialogState.readOnly === true,
    };

    const handleErrorSnackbarOpenClose = (event: SyntheticEvent | MouseEvent, reason?: string) => {
        if (reason === 'clickaway') {
            return;
        }

        setSnackbarError(null);
    };

    return (
        <Box
            sx={{
                display: 'flex',
                boxSizing: 'border-box',
                minHeight: '100vh',
            }}
        >
            <SideBar {...props.sideBarProps}>
                <Box
                    flexShrink={0}
                    sx={{
                        '& .sideMenu': {
                            display: 'flex',
                            flexDirection: 'column',
                            alignItems: 'center',
                            width: '50px',
                        },
                    }}
                >
                    {props.sideMenu}
                </Box>
            </SideBar>
            <Box
                sx={{
                    marginLeft: '50px',
                    display: 'flex',
                    flexDirection: 'column',
                    boxSizing: 'border-box',
                    flexGrow: 1,
                }}
            >
                <Box
                    sx={{
                        zIndex: (theme) => theme.zIndex.appBar + 1,
                        position: 'absolute',
                        width: 0,
                        height: 0,
                        borderStyle: 'solid',
                        borderWidth: '15px 15px 0 0',
                        borderColor: '#e0e0e0 transparent transparent transparent',
                    }}
                />
                <Box
                    sx={{
                        display: 'flex',
                        flexDirection: 'column',
                        boxSizing: 'border-box',
                        flexGrow: 1,
                    }}
                >
                    {props.breadcrumb}
                    <Box
                        sx={{
                            zIndex: 20,
                            top: 64,
                            display: 'flex',
                            height: 15,
                            backgroundImage: 'linear-gradient(to bottom, white, transparent)',
                            width: 'calc( 100% - 66px)',
                            position: 'absolute',
                        }}
                    />
                    <Box
                        sx={{
                            display: 'flex',
                            boxSizing: 'border-box',
                            flex: 1,
                            width: '100%',
                            position: 'relative',
                        }}
                    >
                        <Box
                            height="100%"
                            position="absolute"
                            width="100%"
                            sx={{
                                overflowX: 'auto',
                            }}
                            pt="15px"
                            bgcolor="background.paper"
                            flexShrink={1}
                            flexGrow={1}
                        >
                            {props.children}
                        </Box>
                    </Box>
                </Box>
            </Box>

            <CancelConfirmDialog {...props.cancelConfirmDialogProps} />

            <MainDrawer {...drawerProps}>
                {dialogState.pageComponent !== null && dialogState.type === 'drawer' && (
                    <dialogState.pageComponent {...dialogPageProps} />
                )}
            </MainDrawer>

            {dialogState.pageComponent !== null && (
                <>
                    <InfoDialog {...infoProps} fullscreen={dialogState.type === 'fullscreen'}>
                        {(dialogState.type === 'fullscreen' || dialogState.type === 'info') && (
                            <dialogState.pageComponent {...dialogPageProps} />
                        )}
                    </InfoDialog>

                    <PageDialog {...lightboxProps}>
                        {dialogState.type === 'lightbox' && (
                            <dialogState.pageComponent {...dialogPageProps} />
                        )}
                    </PageDialog>

                    {dialogState.type === 'hidden' && (
                        <dialogState.pageComponent {...dialogPageProps} />
                    )}
                </>
            )}

            <Snackbar
                anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'center',
                }}
                open={snackbarError !== null}
                onClose={handleErrorSnackbarOpenClose}
            >
                <div>
                    {snackbarError !== null && (
                        <GQLError error={snackbarError} onClose={handleErrorSnackbarOpenClose} />
                    )}
                </div>
            </Snackbar>
        </Box>
    );
};

export default LoggedInTemplate;
