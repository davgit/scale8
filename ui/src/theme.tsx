import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
