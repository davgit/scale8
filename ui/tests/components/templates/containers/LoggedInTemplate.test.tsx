import renderer from 'react-test-renderer';
import * as nextRouter from 'next/router';
import LoggedInTemplate from '../../../../src/components/templates/containers/LoggedInTemplate';
import theme from '../../../../src/theme';
import { ThemeProvider } from '@mui/styles';
import { StyledEngineProvider } from '@mui/styled-engine';
import { Theme } from '@mui/system';

declare module '@mui/styles/defaultTheme' {
    // eslint-disable-next-line @typescript-eslint/no-empty-interface
    interface DefaultTheme extends Theme {}
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
nextRouter.useRouter = jest.fn();
(nextRouter.useRouter as jest.Mock).mockImplementation(() => ({ route: '/' }));

jest.mock('@mui/material/Portal', () => '');
// (
//     {
//     render: function () {
//         return <></>;
//     },
//     getMountNode: function () {
//         return null;
//     },
// }));

describe('LoggedInTemplate', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('renders LoggedInTemplate unchanged', () => {
        const tree = renderer
            .create(
                <StyledEngineProvider injectFirst>
                    <ThemeProvider theme={theme}>
                        <LoggedInTemplate
                            breadcrumb={<div />}
                            cancelConfirmDialogProps={{
                                open: false,
                                handleCancel: jest.fn(),
                                handleConfirm: jest.fn(),
                                text: 'something',
                            }}
                            sideBarProps={{
                                handleNotificationClick: jest.fn(),
                                userSelectorProps: {
                                    loading: false,
                                    userName: 'name',
                                    userEmail: 'email',
                                    handleAccountClick: jest.fn(),
                                    handleLogoutClick: jest.fn(),
                                },
                                isAdmin: true,
                                unreadNotifications: 5,
                            }}
                            sideMenu={<div />}
                        >
                            <div>
                                <div />
                            </div>
                        </LoggedInTemplate>{' '}
                    </ThemeProvider>
                </StyledEngineProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
