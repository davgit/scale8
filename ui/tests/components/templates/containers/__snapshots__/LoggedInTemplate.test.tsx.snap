// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInTemplate renders LoggedInTemplate unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-jq8mvv"
  >
    <div
      class="MuiBox-root css-7xm6n4"
    >
      <div
        class="MuiBox-root css-tmox93"
      >
        <svg
          viewBox="0 0 186 339"
          width="27"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 55.03,16.48 C 76.14,-4.63 110.37,-4.63 131.48,16.48 131.48,16.48 93.25,54.69 93.25,54.69 93.25,54.69 55.03,16.48 55.03,16.48 Z"
            id="top-1"
            style="fill: #c6e54b; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 93.25,54.69 C 93.25,54.69 131.66,16.29 131.66,16.29 131.66,16.29 169.93,54.55 169.93,54.55 169.93,54.55 131.52,92.95 131.52,92.95 131.52,92.95 93.25,54.69 93.25,54.69 Z"
            id="top-2"
            style="fill: #fab300; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 169.75,54.74 C 190.86,75.84 190.86,110.06 169.75,131.16 169.75,131.16 131.52,92.95 131.52,92.95 131.52,92.95 169.75,54.74 169.75,54.74 Z"
            id="top-3"
            style="fill: #ffdc51; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 16.25,131.77 C -4.86,110.67 -4.86,76.45 16.25,55.34 16.25,55.34 54.47,93.56 54.47,93.56 54.47,93.56 16.25,131.77 16.25,131.77 Z"
            id="mid-1"
            style="fill: #01c8a1; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 16.25,131.77 C 16.25,131.77 54.65,93.38 54.65,93.38 54.65,93.38 92.92,131.64 92.92,131.64 92.92,131.64 54.52,170.03 54.52,170.03 54.52,170.03 16.25,131.77 16.25,131.77 Z"
            id="mid-2"
            style="fill: #0096a6; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 54.52,170.03 C 54.52,170.03 92.92,131.64 92.92,131.64 92.92,131.64 131.19,169.90 131.19,169.90 131.19,169.90 92.79,208.29 92.79,208.29 92.79,208.29 54.52,170.03 54.52,170.03 Z"
            id="mid-3"
            style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 169.46,208.16 C 169.46,208.16 131.05,246.55 131.05,246.55 131.05,246.55 92.78,208.29 92.78,208.29 92.78,208.29 131.19,169.90 131.19,169.90 131.19,169.90 169.46,208.16 169.46,208.16 Z"
            id="mid-4"
            style="fill: #aa74e6; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 169.46,208.16 C 190.57,229.26 190.57,263.48 169.46,284.59 169.46,284.59 131.24,246.37 131.24,246.37 131.24,246.37 169.46,208.16 169.46,208.16 Z"
            id="mid-5"
            style="fill: #9042e7; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 16.16,284.90 C -4.95,263.80 -4.95,229.58 16.16,208.47 16.16,208.47 54.38,246.69 54.38,246.69 54.38,246.69 16.16,284.90 16.16,284.90 Z"
            id="bottom-1"
            style="fill: #ff0084; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 92.65,284.95 C 92.65,284.95 54.25,323.35 54.25,323.35 54.25,323.35 15.98,285.08 15.98,285.08 15.98,285.08 54.38,246.69 54.38,246.69 54.38,246.69 92.65,284.95 92.65,284.95 Z"
            id="bottom-2"
            style="fill: #fe657a; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
          <path
            d="M 130.87,323.16 C 109.76,344.27 75.54,344.27 54.43,323.16 54.43,323.16 92.65,284.95 92.65,284.95 92.65,284.95 130.87,323.16 130.87,323.16 Z"
            id="bottom-3"
            style="fill: #c63d51; fill-opacity: 1; fill-rule: nonzero; stroke: none; stroke-width: 0.352777;"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root css-16kb41"
      />
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-8ihzru"
        >
          <div />
        </div>
      </div>
      <div
        class="MuiBox-root css-fthphr"
      />
      <div
        class="MuiBox-root css-11mekov"
      >
        <div
          class="MuiBox-root css-0"
        >
          <button
            aria-label="Admin Area"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-h27epy-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
              data-testid="SettingsApplicationsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm7-7H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-1.75 9c0 .23-.02.46-.05.68l1.48 1.16c.13.11.17.3.08.45l-1.4 2.42c-.09.15-.27.21-.43.15l-1.74-.7c-.36.28-.76.51-1.18.69l-.26 1.85c-.03.17-.18.3-.35.3h-2.8c-.17 0-.32-.13-.35-.29l-.26-1.85c-.43-.18-.82-.41-1.18-.69l-1.74.7c-.16.06-.34 0-.43-.15l-1.4-2.42c-.09-.15-.05-.34.08-.45l1.48-1.16c-.03-.23-.05-.46-.05-.69 0-.23.02-.46.05-.68l-1.48-1.16c-.13-.11-.17-.3-.08-.45l1.4-2.42c.09-.15.27-.21.43-.15l1.74.7c.36-.28.76-.51 1.18-.69l.26-1.85c.03-.17.18-.3.35-.3h2.8c.17 0 .32.13.35.29l.26 1.85c.43.18.82.41 1.18.69l1.74-.7c.16-.06.34 0 .43.15l1.4 2.42c.09.15.05.34-.08.45l-1.48 1.16c.03.23.05.46.05.69z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <div
            aria-label="Manage Organizations"
            class="MuiBox-root css-0"
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-x6ov7r-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="BusinessIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            aria-label="Notifications"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-x6ov7r-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiBadge-root css-1c32n2y-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                data-testid="NotificationsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-colorSecondary MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRightRectangular css-1n4jbui-MuiBadge-badge"
              >
                5
              </span>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-1pg8bi"
        />
        <button
          aria-controls="simple-menu"
          aria-haspopup="true"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pk0057-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-14fb8r7-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-udmot3-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiBox-root css-uanaa8"
        />
      </div>
    </div>
    <div
      class="MuiBox-root css-fq0scu"
    >
      <div
        class="MuiBox-root css-15i9l62"
      />
      <div
        class="MuiBox-root css-1qy9df7"
      >
        <div />
        <div
          class="MuiBox-root css-19b5z3n"
        >
          <div
            class="MuiBox-root css-yzu9ne"
          />
        </div>
        <div
          class="MuiBox-root css-rvm16z"
        >
          <div
            class="MuiBox-root css-a744pu"
          >
            <div>
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
</DocumentFragment>
`;
